lib/modules/txt.mjs

```diff
@@ -343,3 +343,43 @@ export async function validateCopyrightNoticeSectionIsNumbered (doc, { mode = MO
 
   return result
 }
+
+/**
+ * Validate if document meets specific structural requirements
+ *
+ * @param {Object} doc Document to validate
+ * @param {Object} [opts] Additional options
+ * @param {number} [opts.mode=0] Validation mode to use
+ * @returns {Array} List of errors/warnings or empty if fully valid
+ */
+export async function validateTableOfContentsAndDocumentPages (doc, { mode = MODES.NORMAL } = {}) {
+  const result = []
+
+  const RFC_DRAFT_MAX_PAGES = 15
+  const lacksTableOfContents = !doc.data.possibleIssues.isTableOfContentsExists
+  const isTooLong = doc.data.pageCount - 1 > RFC_DRAFT_MAX_PAGES
+
+  if (doc.data.possibleIssues.isTableOfContentsExists && RFC_DRAFT_MAX_PAGES >= doc.data.pageCount - 1) return result
```

Comment:
why is this not written using the two variables computed above instead of restating them? (even if there was a < and <= difference in the comparison in idnits2 there shouldn't be in idnits3).